src/ResourceManager/NotificationHubs/Commands.NotificationHubs/Commands/NotificationHub/SetAzureNotificationHubAuthorizationRules.cs

```diff
@@ -0,0 +1,90 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Common.Authentication;
+using Microsoft.Azure.Commands.NotificationHubs.Models;
+using Microsoft.Azure.Management.NotificationHubs.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Linq;
+using System.IO;    
+
+namespace Microsoft.Azure.Commands.NotificationHubs.Commands.NotificationHub
+{
+
+    [Cmdlet(VerbsCommon.Set, "AzureRmNotificationHubAuthorizationRules"), OutputType(typeof(SharedAccessAuthorizationRuleAttributes))]
+    public class SetAzureNotificationHubAuthorizationRules : AzureNotificationHubsCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The name of the resource group")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroup { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = "Namespace Name.")]
+        [ValidateNotNullOrEmpty]
+        public string Namespace { get; set; }
+
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            HelpMessage = "NotificationHub Name.")]
+        [ValidateNotNullOrEmpty]
+        public string NotificationHub { get; set; }
+
+        [Parameter(Mandatory = true,
+            Position = 3,
+            ParameterSetName = InputFileParameterSetName,
+            HelpMessage = "File name containing a single AuthorizationRule definition.")]
+        [ValidateNotNullOrEmpty]
+        public string InputFile { get; set; }
+
+        [Parameter(Mandatory = true,
+            Position = 3,
+            ParameterSetName = SASRuleParameterSetName,
+            HelpMessage = "NotificationHub AuthorizationRule Object.")]
+        [ValidateNotNullOrEmpty]
+        public SharedAccessAuthorizationRuleAttributes SASRule { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            if (!string.IsNullOrEmpty(ResourceGroup) && !string.IsNullOrEmpty(Namespace) && !string.IsNullOrEmpty(NotificationHub))
+            {
+                if (!string.IsNullOrEmpty(ResourceGroup) && !string.IsNullOrEmpty(Namespace))
```

Comment:
Is there any use of checking  ResourceGroup and Namespace nor null or empty again? You've just checked it on line 68.